erp-playbook/library/mr_provisioner_power.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python
+
+import json
+import requests
+import time
+from datetime import datetime, timedelta
+
+from future.standard_library import install_aliases
+install_aliases()
+
+from urllib.parse import urljoin  # nopep8
+from urllib import quote  # nopep8
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: mr-provisioner-power
+
+short_description: Wait for power on/off.
+
+description:
+    - Wait for power on/off.
+    - Suppot to force power on/off when timeout reached.
+
+options:
+    machine_name:
+        description: Machine name
+        required: true
+    url:
+        description: url to provisioner instance in the form of http://172.27.80.1:5000/
+        required: true
+    token:
+        description: Mr. Provisioner auth token
+        required: true
+    wait_for_power:
+        description: Desired power state
+        required: true
+    timeout:
+        description: Maximum number of seconds before interrupt request.
+        required: false
+        default: 21600
+    sleep:
+        description:  Number of seconds to sleep between checks.
+        required: false
+        default: 300
+    force_power:
+        description: Foce power on/off
```

Comment:
Should this be set to "true/false" or "on/off"? We already know what the desired power state should be using wait_for_power, so we could just have this be boolean (it might be less confusing).

Reply:
Indeed, using a boolean is less confusing. Done.